### PR TITLE
CollectionWidget: Make disabled items available for AT

### PR DESCRIPTION
**Pull request by marker-dao**

### CODE REVIEW

**Review by Zedwag — DISMISSED** [review]

**Review by EugeniyKiyashko — DISMISSED** [review]

File: js/ui/collection/ui.collection_widget.base.js [hunk 1]
```diff
@@ -282,6 +282,10 @@ const CollectionWidget = Widget.inherit({
         this._updateFocusedItemState($target, false);
     },
 
+    _findActiveTarget($el) {
```
Comment (ksercs) [inline js/ui/collection/ui.collection_widget.base.js]: It's not a good practice to shorten a word
https://google.github.io/styleguide/jsguide.html#naming-rules-common-to-all-identifiers
```suggestion
    _findActiveTarget($element) {
```

File: js/ui/collection/ui.collection_widget.base.js [hunk 2]
```diff
@@ -408,7 +416,10 @@ const CollectionWidget = Widget.inherit({
         this._updateFocusedItemState($target, true);
         this.onFocusedItemChanged(this.getFocusedItemId());
 
-        if(this.option('selectOnFocus')) {
+        const { selectOnFocus } = this.option();
+        const isDisabled = this._isDisabled($target);
```
Comment (ksercs) [inline js/ui/collection/ui.collection_widget.base.js]: I see "isDisabled" -> I think about disabled state of the component -> I understand that it's unlikely that we check disabled state of the component -> I go to the variable initialization to check what is really disabled here.
Let's just make the name more complete?
```suggestion
        const isTargetDisabled = this._isDisabled($target);
```

File: testing/testcafe/tests/editors/list/common.ts [hunk 3]
```diff
@@ -314,9 +314,9 @@ test('Disabled item should not have focus (T815151)', async (t) => {
     .click(searchInput)
     .pressKey('tab')
     .expect(firstItem.isFocused)
-    .notOk()
```
Comment (ksercs) [inline testing/testcafe/tests/editors/list/common.ts:317]: We updated asserts but didn't updated the test name.
I suggest to rename this way:
```javascript
test('Disabled item should be focused on tab press to match accessibility criteria', async (t) => {
```

File: testing/tests/DevExpress.ui/collectionWidget.tests.js [hunk 4]
```diff
@@ -1624,10 +1624,13 @@ module('keyboard navigation', {
 
         $element.focusin();
         $item.trigger('dxpointerdown');
+
         this.clock.tick();
+
         keyboard.keyDown('right');
-        assert.equal(instance.option('selectedIndex'), 3, 'selectedIndex is correctly');
-        $item = $($items.get(3));
+        assert.equal(instance.option('selectedIndex'), 0, 'selectedIndex is correctly');
```
Comment (ksercs) [inline testing/tests/DevExpress.ui/collectionWidget.tests.js]:
```suggestion
        assert.strictEqual(instance.option('selectedIndex'), 0, 'selectedIndex is correct');
```

File: testing/tests/DevExpress.ui/collectionWidgetItem.tests.js [hunk 5]
```diff
@@ -151,3 +151,13 @@ QUnit.test('disabled should be rendered correctly after value changed', function
...
+
+QUnit.test('aria-disabled attribute should be set if the item\'s disabled prop is true', function(assert) {
+    const widget = new CollectionWidget($('<div>'), {
+        items: [{ disabled: true }]
+    });
+
+    const $item = widget.itemElements().eq(0);
+
+    assert.ok($item.attr('aria-disabled'), 'element has aria-disabled attribute');
+});
```
Comment (ksercs) [inline testing/tests/DevExpress.ui/collectionWidgetItem.tests.js:163]: Don't we want to test this case for a `disabled` property runtime change?

File: js/ui/collection/ui.collection_widget.base.js (hunk 1 above)
Comment (marker-dao) [reply js/ui/collection/ui.collection_widget.base.js]: Okay, let's create our own style guide) I fixed it in this commit [25bf9f775185867becc632f8384161cc5d472868].

File: js/ui/collection/ui.collection_widget.base.js (hunk 2 above)
Comment (marker-dao) [reply js/ui/collection/ui.collection_widget.base.js]: Yes, a good point! It was fixed here [25bf9f775185867becc632f8384161cc5d472868].

File: testing/testcafe/tests/editors/list/common.ts (hunk 3 above)
Comment (marker-dao) [reply testing/testcafe/tests/editors/list/common.ts:317]: A nice suggestion! Thank you a lot! Was fixed here [25bf9f775185867becc632f8384161cc5d472868].

File: testing/tests/DevExpress.ui/collectionWidget.tests.js (hunk 4 above)
Comment (marker-dao) [reply testing/tests/DevExpress.ui/collectionWidget.tests.js]: Test was updated in this commit [25bf9f775185867becc632f8384161cc5d472868]. Thanks!

File: testing/tests/DevExpress.ui/collectionWidgetItem.tests.js (hunk 5 above)
Comment (marker-dao) [reply testing/tests/DevExpress.ui/collectionWidgetItem.tests.js:163]: Okay, let's do it! I added a new test in this commit [25bf9f775185867becc632f8384161cc5d472868].

File: js/ui/collection/ui.collection_widget.base.js [hunk 6]
```diff
@@ -400,6 +404,10 @@ const CollectionWidget = Widget.inherit({
         }
     },
 
+    _isDisabled($el) {
```
Comment (ksercs) [inline js/ui/collection/ui.collection_widget.base.js]: Here is the same naming issue I mentioned above :)

Comment (marker-dao) [reply js/ui/collection/ui.collection_widget.base.js]: Sorry, wanna sleep :D Fixed here [ef0c18fa47f75bf7ad928610beee9263719f13ab].

**Review by EugeniyKiyashko — DISMISSED** [review]

**Review by ksercs — DISMISSED** [review]

**Review by ksercs — APPROVED** [review]

**Pull request merged**